src/main/java/seedu/address/model/person/History.java

```diff
@@ -1,127 +1,325 @@
+     * @param date The date and time of the appointment to remove.
+     */
+    public void removeAppointments(LocalDateTime date) {
+        if (appointments.contains(date)) {
+            appointments.remove(date);
+        }
+    }
 
     /**
-     * Adds an appointment to the database with the specified content.
+     * Checks if the patient has duplicate appointments for the same date.
      *
-     * @param dateTime Date & time of the appointment.
-     * @param patientId Id of patient in the appointment.
-     * @param doctorId Id of doctor in the appointment.
-     * @return True if appointment was successfully added, false if otherwise.
-     */
-    public boolean addAppointment(LocalDateTime dateTime, Id patientId, Id doctorId, String remarks) {
-        // TODO something new Appointment(dateTime, patientId, doctorId, remarks)
-        // TODO also need add to this.personAppointments
-        // Other notes: check for duplicate appointments / clashing timeslots w doctor & patient
-        // What to throw if got error
+     * @param date The date to check for duplicates.
+     * @return true if duplicate appointments are found, false otherwise.
+     */
+    public boolean checkDuplicateAppointments(LocalDateTime date) {
+        if (Collections.frequency(appointments, date) > 1) {
+            return true; // Duplicate found
+        }
         return false;
     }
 
     /**
-     * Deletes the specified appointment with the respective details.
-     * @param dateTime Date & time of the appointment to delete.
-     * @param patientId Id of the patient in the appointment.
-     * @param doctorId Id of doctor in the appointment.
-     * @return True if appointment was successfully deleted, false if otherwise.
+     * Retrieves the details of a single appointment based on the date, patient, and doctor.
+     *
+     * @param date The date and time of the appointment.
+     * @param patientId The ID of the patient.
+     * @param doctorId The ID of the doctor.
+     * @return The appointment details.
+     * @throws AppNotFoundException if no appointment is found for the given patient and doctor.
      */
-    public static boolean deleteAppointment(LocalDateTime dateTime, Id patientId, Id doctorId) {
-        // TODO something
-        // TODO rmb need delete from this.personAppoitments
-        return false;
+    public Appointment getOneAppointmentDetail(LocalDateTime date, Id patientId, Id doctorId)
+            throws AppNotFoundException {
+        if (!appointments.contains(date)) {
+            throw new AppNotFoundException("No appointment found in this patient's history for the given date.");
+        }
+        Appointment appointment = appointmentDatabase.get(date);
+        if (appointment == null || !appointment.getPatientId().equals(patientId)
+                || !appointment.getDoctorId().equals(doctorId)) {
+            throw new AppNotFoundException("No such appointment found for the given patient and doctor.");
```

Comment:
As there may be multiple appointments on the same LocalDateTime but with different patients / doctors, I am wondering if this may accidentally appointmentDatabase.get an appointment on that date but for a different doctor & patient